### ISSUE TITLE
rd.neednet=1 should be a soft requirement

### ISSUE THREAD

**Issue by mudler** [rancher/elemental-toolkit#388]
**Is your feature request related to a problem? Please describe.**
Currently, if we fail to get an IP via DHCP during initramfs, we drop to the emergency shell. To boot up the system is sufficient to type "exit"

**Describe the solution you'd like**
The network error shouldn't prevent the full booting of the system, we use `rd.neednet=1` by default as boot params which enables this behavior. The system might be configured also via cloud-init to set ip addresses, so we shouldn't assume we are always getting one (and we have network access). Ideally I would see a default option to be "permissive" but optionally that could be set in "enforced" (like SELinux, for e.g.)

**Describe alternatives you've considered**
Keep failing hardly. This might be a good idea for the Cloud, as without any IP assigned, we might not want to have workload to start.

**Additional context**

![Screenshot from 2021-07-08 10-20-13](https://user-images.githubusercontent.com/2420543/124895537-ebfda580-dfdc-11eb-89e1-6c7c1ed03ee4.png)
![Screenshot from 2021-07-08 10-22-56](https://user-images.githubusercontent.com/2420543/124895542-ed2ed280-dfdc-11eb-95a7-d9f16952c6f6.png)

Relates to #7

**Comment by davidcassany** [comment]
I did a quick search about this topic and it looks like it might not be simple to fix without making an explicit choice on the network management system. In 15.3 we have `network-legacy` and `network-manager` dracut modules, and in TW we have an extra `network-wicked`. 15.3 series default into `network-legacy` and TW defaults to `network-wicked`. I did a quick search and I am not sure we can find a solution that is not tight to any specific dracut module.
